### PR TITLE
Rework appveyor

### PR DESCRIPTION
**Pull request by Taluu**
And try to test it on multiple php versions too. I also added php 7.2 to travis while I was at it.

Mostly inspired from @Majkl578 and @photodude work on [doctrine dbal's appveyor config](https://github.com/doctrine/dbal/blob/master/.appveyor.yml)

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -16,31 +19,49 @@ skip_commits:
...
+
+            if (!(Test-Path c:\tools\composer\composer.phar)) {
+                appveyor-retry appveyor DownloadFile https://getcomposer.org/composer.phar -Filename C:\tools\composer\composer.phar
+                Set-Content -path 'C:\tools\composer\composer.bat' -Value ('@php C:\tools\composer\composer.phar %*')
+            }
+        }
+
     - cd c:\projects\behat
-    - composer install --no-progress --ansi
+    - rm composer.lock
```
Comment (Majkl578) [inline appveyor.yml]: You don't need this one.

Comment (Taluu) [reply appveyor.yml]: yeah wasn't sure if some cache would not kick in. Removed.

**Comment by Majkl578** [comment]
Don't trust metrics here, cache is never persisted for PRs. If you want to benchmark (it will always vary due to load etc.), you will need to test it in the repository itself (branch is enough), not through PR. :)

**Comment by Majkl578** [comment]
You can also check Doctrine DBAL build times here: https://ci.appveyor.com/project/doctrine/dbal
(The config is referenced above and is a bit more more complicated than yours, also collects and sends coverage report.)

**Comment by Taluu** [comment]
> However, this will extend the build time from ~2.5min to ~17.5min. Is it worth it? We already run tests against all supported php versions on travis. Appveyor is ther just so we confirm there's no incompatibility issues with windows. Latest PHP version should be sufficient imo.

Yeah, been wondering that. Thought some cache could kick in, but then we could indeed at least test only the latest version... or maybe 5.6 and 7.2 ?

**Comment by Majkl578** [comment]
@Taluu Btw you will probably need different directories for different PHP versions in order to make cache work properly.

File: appveyor.yml [hunk 2]
```diff
@@ -16,31 +19,51 @@ skip_commits:
     message: /\[ci skip\]/
 
 cache:
-    - c:\php -> appveyor.yml
+    - C:\ProgramData\chocolatey\bin -> appveyor.yml
+    - C:\ProgramData\chocolatey\lib -> appveyor.yml
+    - C:\tools\php5.6 -> appveyor.yml
```
Comment (Majkl578) [inline appveyor.yml]: If you make the version a subdirectory of "C:\tools\php", this and other occurrences may not turn that ugly. :)

**Comment by Taluu** [comment]
I think I'm gonna test only on latest php version as @jakzal suggested, as it's a mess with the env vars. It seems that chocolatey doesn't properly install when env vars are involved...

**Comment by jakzal** [comment]
Looks good to me 👍

File: .travis.yml [hunk 3]
```diff
@@ -20,7 +20,7 @@ matrix:
       env: SYMFONY_VERSION='2.7.*'
     - php: 5.6
       env: SYMFONY_VERSION='2.8.*'
-    - php: 7.1
+    - php: 7.2
       env: SYMFONY_VERSION='4.0.*@dev'
```
Comment (stof) [inline .travis.yml]: the one which should be tested explicitly here is the 3.4 LTS, like done for 2.7 and 2.8. The `4.0.*@dev` is useless

File: appveyor.yml [hunk 4]
```diff
@@ -16,31 +16,48 @@ skip_commits:
     message: /\[ci skip\]/
 
 cache:
-    - c:\php -> appveyor.yml
+    - C:\ProgramData\chocolatey\bin -> appveyor.yml
+    - C:\ProgramData\chocolatey\lib -> appveyor.yml
+    - C:\tools\php -> appveyor.yml
+    - C:\tools\composer -> appveyor.yml
+    - '%LOCALAPPDATA%\Composer\files -> composer.json'
```
Comment (stof) [inline appveyor.yml]: no need to expire them when the composer.json is changing. Even if we change one dependency, the others could still benefit from the cache.

**Comment by Taluu** [comment]
Changes done, as @stof suggested them. I rebase against the master too while I was at it.

**Pull request merged**